Core/Rewriters/Strategies/ArgListEqualArgumentArgumentRewriteStrategy.cs

```diff
@@ -0,0 +1,35 @@
+﻿//  Copyright (c) rubicon IT GmbH
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using RhinoMocksToMoqRewriter.Core.Utilities;
+
+namespace RhinoMocksToMoqRewriter.Core.Rewriters.Strategies
+{
+  public class ArgListEqualArgumentArgumentRewriteStrategy : IArgumentRewriteStrategy
+  {
+    public ArgumentSyntax Rewrite (ArgumentSyntax node)
+    {
+      var implicitArrayCreationNode = node.DescendantNodes()
+          .FirstOrDefault (
+              s => s.IsKind (SyntaxKind.ImplicitArrayCreationExpression) && s.Parent.IsKind (SyntaxKind.Argument)) as ImplicitArrayCreationExpressionSyntax;
+
+      return MoqSyntaxFactory.ImplicitArrayCreationArgument (implicitArrayCreationNode!.Initializer)
+          .WithLeadingTrivia (node.GetLeadingTrivia());
```

Comment:
This only handles `new[]{...}`, but what about `List<>`? Shouldn't this just extract the argument passend into `Equal(...)` and return it? Don't forget to add test cases.